### ISSUE TITLE
multiext support for named output

### ISSUE THREAD

**Issue by fgvieira** [snakemake/snakemake#1335]
**Is your feature request related to a problem? Please describe.**

Right now it is not possible to cache a rule's output when there are several named output files. With several output files we can use `multiext`, but it does not allow for named output.

**Describe the solution you'd like**

I was thinking that it would be nice to have a way to have multiple named output files with `multiext`, for example:

```
output:
    multiext("path/to/output_file.", out1 = "ext1",  out2 = "ext2")
```

or:
```
output:
    out1, out2 = multiext("path/to/output_file.", "ext1", "ext2")
```

**Comment by notestaff** [comment]
I'd like to second the request for this feature.

**Comment by johanneskoester** [comment]
Yes, this should be possible. I like the syntax of the first suggestion more. This would have to be implemented in snakemake.rules.Rule._set_inoutput_item and snakemake.io.multiext.
